block/base/mempool.go

```diff
@@ -121,49 +136,63 @@ func (cm *Mempool[C]) Contains(tx sdk.Tx) bool {
 	return ok
 }
 
-// Compare determines the relative priority of two transactions belonging in the same lane.
+func (cm *Mempool[C]) Compare(ctx sdk.Context, this, other sdk.Tx) (int, error) {
+	return cm.comparator(ctx, this, other)
+}
+
+// ComparePriorityNonce determines the relative priority of two transactions belonging in the same lane.
 // There are two cases to consider:
 //  1. The transactions have the same signer. In this case, we compare the sequence numbers.
 //  2. The transactions have different signers. In this case, we compare the priorities of the
 //     transactions.
 //
 // Compare will return -1 if this transaction has a lower priority than the other transaction, 0 if
 // they have the same priority, and 1 if this transaction has a higher priority than the other transaction.
-func (cm *Mempool[C]) Compare(ctx sdk.Context, this sdk.Tx, other sdk.Tx) (int, error) {
-	signers, err := cm.extractor.GetSigners(this)
-	if err != nil {
-		return 0, err
-	}
-	if len(signers) == 0 {
-		return 0, fmt.Errorf("expected one signer for the first transaction")
-	}
-	// The priority nonce mempool uses the first tx signer so this is a safe operation.
-	thisSignerInfo := signers[0]
+func PriorityNonceComparator(extractor signer_extraction.Adapter, txPriority TxPriority[string]) Comparator {
```

Comment:
I would maybe add a disclaimer that this doesnt work well with multiple fees in a single lane.